### PR TITLE
[AUTOMATE-832] Use db_id in iam_statement_projects

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

Use project's `db_id` in the `iam_statement_projects` table instead of `id`

### :athletic_shoe: Demo Script / Repro Steps

Setup:

```
start_all_services
chef-automate iam upgrade-to-v2 --beta2.1
export TOK=`chef-automate iam token create my-token --admin`
```

Make projects:

```
curl -kH "api-token: $TOK" -XPOST -d "$(jq -n '{ id: "project1", name: "my project 1" }')" https://localhost/apis/iam/v2beta/projects
curl -kH "api-token: $TOK" -XPOST -d "$(jq -n '{ id: "project2", name: "my project 2" }')" https://localhost/apis/iam/v2beta/projects
```

Make file `policy.json`:

```
{
  "name": "Test Policy 1",
  "id": "test-policy-1",
  "members": [ "user:local:*", "token:*" ],
  "statements": [
    {
      "effect": "ALLOW",
      "actions": [ "iam:*" ],
      "projects": [ "project1", "project2" ]
    }
  ]
}
```

Create policy:

```
curl -kH "api-token: $TOK" -XPOST -d @policy.json https://localhost/apis/iam/v2beta/policies
```

Run migration:

```
rebuild components/authz-service
```

`project_id` now refers to project's `db_id`:

```
chef_authz_service=# select * from iam_statement_projects;
             statement_id             | project_id
--------------------------------------+------------
 9e7fefc0-c7dc-40e6-b807-9a4a56f81cf2 |          1
 e7486529-2cff-42da-8803-040f0e05f513 |          1
 04f39a5d-b672-4f01-a53e-95e172ed11cc |          1
 a174a052-6ca4-4999-8cd7-47982830a08d |          1
 d0c16669-4254-4a60-8617-2dbdf0e97b3f |          1
 756a0faa-9b5f-4ee0-a026-fd55bbfee509 |          5
 756a0faa-9b5f-4ee0-a026-fd55bbfee509 |          6
 6506d801-393b-445d-9181-1eff4f97fdb7 |          2
(8 rows)
```

### :chains: Related Resources

- #832 

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/authz-service/storage/postgres/migration/sql/README.md [hunk 1]
```diff
@@ -57,3 +57,4 @@
 - [`55_use_project_db_id_in_references_of_project_roles.up.sql`](55_use_project_db_id_in_references_of_project_roles.up.sql)
 - [`56_add_db_id_statements.up.sql`](56_add_db_id_statements.up.sql)
 - [`57_use_project_db_id_in_references_of_policies.up.sql`](57_use_project_db_id_in_references_of_policies.up.sql)
+- [`58_use_project_db_id_in_references_of_projects.up.sql`](58_use_project_db_id_in_references_of_projects.up.sql)
```
Comment (srenatus) [inline components/authz-service/storage/postgres/migration/sql/README.md]: That name is ambiguous -- let's go with 
```suggestion
- [`58_use_project_db_id_in_references_of_policy_statements.up.sql`](58_use_project_db_id_in_references_of_policy_statements.up.sql)
```

File: components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql [hunk 2]
```diff
@@ -0,0 +1,159 @@
+BEGIN;
+
+ALTER TABLE iam_statement_projects
+    DROP CONSTRAINT iam_statement_projects_project_id_fkey;
+
+ALTER TABLE iam_statement_projects RENAME COLUMN project_id TO project_temp_id;
+
+ALTER TABLE iam_statement_projects ADD COLUMN project_id SERIAL;
```
Comment (blakestier) [inline components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql]: Prob don't want `SERIAL` bc we're manually populating it

File: components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            pol.type,
+            -- get policy's statements using temporary table
+            ( WITH statement_rows AS (
+                    SELECT
+                        stmt.id,
+                        stmt.effect,
+                        stmt.actions,
+                        stmt.resources,
+                        stmt.role,
+                        -- get each statement's projects by cross-referencing iam_policy_statements and iam_statement_projects
```
Comment (blakestier) [inline components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql]: we did this in other migrations, too, but this comment is invalid bc we nixed `iam_policy_statements`

File: components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql (hunk 2 above)
Comment (srenatus) [reply components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql]: Isn't this even kind of wrong? 😉 Should be INTEGER, I suppose. There's nothing _serial_ about it; it's not auto-incrementing ... (ref: https://www.postgresql.org/docs/9.1/datatype-numeric.html)

Comment (tylercloke) [reply components/authz-service/storage/postgres/migration/sql/58_use_project_db_id_in_references_of_projects.up.sql]: Looks like we did that wrong elsewhere as well. I'll fix here for starters.

**Pull request merged**